framework/include/kernels/KernelBase.h

```diff
@@ -73,6 +73,12 @@ class KernelBase :
   /// Computes d-residual / d-jvar... storing the result in Ke.
   virtual void computeOffDiagJacobian(unsigned int jvar) = 0;
 
+  /// Compute this Kernel's contribution to the diagonal Jacobian entries
+  virtual void computeNonlocalJacobian() = 0;
```

Comment:
Any reason you declare this as pure virtual? I makes it necessary to add an empty implementation in _every_ class derived from `KernelBase`.